### ISSUE TITLE
Convert unittest class base tests to pytest function tests

### ISSUE THREAD

**Issue by terencehonles** [jazzband/django-redis#521]
By converting unittest class base tests to pytest function based tests we can use some nifty features of pytest such as fixtures* and parameterization :tada: 

See: https://docs.pytest.org/en/stable/unittest.html#pytest-features-in-unittest-testcase-subclasses

Example of what needs to be done: #520

**Comment by WisdomPill** [comment]
It has been merged now, as a matter of fact we could split `test_backend.py` unittest classes in it's own module (namely `backend`) to have smaller files, separate testing different aspects of the library and proceed by steps.

I think a test of one part of backend might simplify contributing.

@terencehonles do you agree?

**Comment by terencehonles** [comment]
Definitely, the classes may make sense to be how things are split up (I haven't looked at how they are organized), and if someone needs to share fixtures between tests they can add the file `conftest.py` in the tests package and put the fixtures there. When placed there they will be requestable by any modules at or below that level.

**Comment by joonyoungleeduke** [comment]
I can try working on this!
